JustSaying/AwsTools/MessageHandling/INonRetryable.cs

```diff
@@ -0,0 +1,6 @@
+namespace JustSaying.AwsTools.MessageHandling
+{
+#pragma warning disable CA1040
+    public interface INonRetryable { }
```

Comment:
Could we use an attribute rather than using marker interfaces (which is what this suppressed warning is trying to coerce the design towards)?

Reply:
This would be considerably more convoluted from a reflection perspective. 
I guess it would be a performance penalty only for the failed messages. But could make an impact in high rate noisy (i.e. high rate of retry) environment where handling the message is fast. There are many reasons why you may wish to avoid marker interfaces not sure this is one of them. 
Personally I think it is this or the strategy you mention below. Checking the types custom attributes for a specific attribute in this application feels a bit expensive and not knowing the life time or scope of the dispatcher means it would be hard and of diminishing returns to cache the cost of checking for attributes to improve performance.